succ ver

abc982627271 opened the pull request:
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

abc982627271 committed:
succ version

ZiheLiu committed:
[Enhancement] Support pipeline level shuffle for CTE (#9426)

meegoo committed:
[Enhancement] Remove broker load unnecessary cast (#9158)

hellolilyliuyi committed:
delete-execute-as-in-release-note (#9449)

liuyehcf committed:
[BugFix] Fix error of ScanTime (#9448)

Pslydhh committed:
[feature] Add black list for insert (#9343)

Add blacklist for insert stmt:

admin set frontend config ("enable_sql_blacklist" = "true")
ADD SQLBLACKLIST "insert into .+ values.+";
then, execute
insert into action333 values(333, 'ddd', '2020-02-25 17:33:22');
ERROR 1064 (HY000): Access denied; This sql is in blacklist, please contact your admin

chaoyli committed:
[Refactor] Removed not used log upon start and unified drop_tablet_unlock interface (#9318)

Many logs upon start is non-sense and the pull request remove these logs.
The AlterTask is already not used and drop_tablet_unlocked can be unified.

TszKitLo40 committed:
[Lake] Support export table (#9307)

adzfolc committed:
Fix code typos (#9459)

stdpain committed:
[Bugfix] fix invalid dictionary in schema change (#9401)

adzfolc committed:
Fix code typos (#9466)

xiaoyong-z committed:
[Feature] Support export without broker (#9171)

Astralidea committed:
Upgrade mysql-connector-java due to security (#9468)

wyb committed:
[BugFix] Use out exprs column as spark rdd schema (#9437)

After #8578 fix, the chunk and arrow batch columns order is already base on output exprs.
Spark uses selected columns as rdd schema, so generate selected columns from output exprs.

nshangyiming committed:
[Enhancement] keep tablet scheduling related config naming consistent (#9137)

Add a new alias annotation method to keep forward compatibility in ConfigBase class.
Rename and rearrange some configurations in Config.java.

femiiii committed:
[Doc]delete content that prepare for 2.4 (#9457)

* [Doc]delete content that prepared for 2.4

* modifications

* Update REVOKE.md

nshangyiming committed:
[Enhancement] lower the log level when add replica on tablet report (#9476)

harveyyue committed:
[SR SQL Planner Summer Camp] New Parser and Analyzer support schema change (column change) operation-related statements (#7941) (#9123)

wyb committed:
[Lake] support delete from lake table (#9352)

1. FE creates DeletePredicatePB which is used in BE tablet metadata directly.
2. No longer use the old push task api.

abc982627271 committed:
[Lake] Support alter table drop partition (#9101)

mxdzs0612 committed:
Iceberg support unknown type (#9431)

Related to #8881 
When there are unsupported types in an Iceberg table, exception will not be thrown if users don't select columns with such types. Like the followings:
```sql
hive> select * from t.m;
OK
1	{"a":"b"}
MySQL [(none)]> select * from ice.t.m;
ERROR 1064 (HY000): External Table Column [m] convert failed, and column type is known!
MySQL [(none)]> select i from ice.t.m;
+------+
| i    |
+------+
|    1 |
+------+
1 row in set (0.08 sec)
```

xiaoyong-z committed:
[Refactor] Broker error msg (#9484)

Signed-off-by: xyz <a997647204@gmail.com>

Youngwb committed:
[BugFix] Fix correlated subquery unknown error when equals predicate has expression (#9438)

When correlated subquery sql like 
```
select t0.v1 from t0 where (t0.v2 in (select t1.v4 from t1 where t0.v3 + t1.v5 = 1)) is NULL
```

The QuantifiedApply2OuterJoinRule would transfrom plan like 
```
  /*
         * In:
         * before: select t0.v1, t0.v2 in (select t1.v2 from t1 where t0.v3 = t1.v3) from t0;
         * after: with xx as (select t1.v2, t1.v3 from t1)
         *        select t0.v1,
         *             case
         *                 // t1 empty table and without where clause `t0.v3 = t1.v3`, then t1Rows may be 0
         *                 // t1 empty table, if t1Rows is null, means the result of join correlation predicate must be false
         *                 // `any_or_null in (empty) -> false`
         *                 when t1Rows = 0 or t1Rows is null then false
         *                 // `null in (any_or_null...) -> null`
         *                 when t0.v2 is null then null
         *                 // `a in (a, [any_or_null...]) -> true`
         *                 when t1d.v2 is not null then true
         *                 // `a in (null, [not_a_or_null...]) -> null`
         *                 when v2NotNulls < t1Rows then null
         *                 // `a in (not_a...) -> false`, not_a cannot be null
         *                 else false
         *             end
         *         from t0
         *              left outer join (select xx.v2, xx.v3 from xx group by xx.v2, xx.v3) as t1d
         *                              on t0.v2= t1d.v2 and t0.v3 = t1d.v3
         *              left outer join (select v3, count(*) as t1Rows, count(xx.v2) as v2NotNulls from xx group by xx.v3) as t1c
         *                              on t1c.v3 = t0.v3
         *
         *                                           CTEAnchor
         *                                          /        \
         *                                    CTEProduce     Project
         *    Apply(t0.v2 in t1.v2)              /               \
         *    /          \          ===>     Project         left/cross join
         *   t0          t1                   /              /          \
         *                                Filter         Left join       Agg(count)
         *                                 /             /       \           \
         *                               t1            t0     Agg(distinct)  CTEConsume
         *                                                         \
         *                                                        CTEConsume
         * */
```

The agg(distinct) and agg(count) should add the v5 column to group by which will used in left join/cross join predicate

Youngwb committed:
[Enhancement] improve join reorder greedy algorithm (#9070)

dirtysalt committed:
[BugFix] fix empty scan ranges in ConnectorScanNode (#9458)

ZiheLiu committed:
[Enhancement] Set cpu_limit of default workgroup to num_executor_threads (#9478)

trueeyu committed:
No need use write lock for base tablet (#9397)

No need use write lock for base tablet

wyb commented on fe/fe-core/src/main/java/com/starrocks/persist/PartitionPersistInfoV2.java:
```diff
@@ -29,6 +29,8 @@ public class PartitionPersistInfoV2 implements Writable {
     @SerializedName("isTempPartition")
     private boolean isTempPartition;
+    @SerializedName("StoarageInfo")
+    private com.starrocks.lake.StorageInfo storageInfo;
```
Better to import `com.starrocks.lake.StorageInfo`.

wyb commented on fe/fe-core/src/main/java/com/starrocks/persist/AddPartitionsInfoV2.java:
```diff
@@ -0,0 +1,40 @@
+import java.io.IOException;
+import java.util.List;
+
+public class AddPartitionsInfoV2 implements Writable {
```
use json to serialize.

wyb commented on fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java:
```diff
@@ -33,8 +33,8 @@ public static void checkTableExist(Database db, String tableName) throws DdlExce
 
     // check table type is OLAP
     public static void checkTableTypeOLAP(Database db, Table table) throws DdlException {
```
```suggestion
    public static void checkNativeTable(Database db, Table table) throws DdlException {
```

wyb replied:
https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/catalog/Table.java#L55

wyb commented on fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:
```diff
@@ -1069,40 +1071,70 @@ private void addRangePartitionLog(Database db, OlapTable olapTable, List<Partiti
-                    isTempPartition);
-            editLog.logAddPartition(info);
+            if (olapTable.isLakeTable()) {
+                PartitionPersistInfoV2 info = new RangePartitionPersistInfo(db.getId(), olapTable.getId(), partition,
+                        partitionDescs.get(0).getPartitionDataProperty(), partitionInfo.getReplicationNum(partition.getId()),
+                        partitionInfo.getIsInMemory(partition.getId()), isTempPartition,
+                        ((RangePartitionInfo) partitionInfo).getRange(partition.getId()),
+                        ((SingleRangePartitionDesc) partitionDescs.get(0)).getStorageInfo());
+                editLog.logAddPartition(info);
```
Use `logAddPartitions` even if add one partition.

liuyehcf committed:
[Enhancement] Refine profile mechanism (#9477)

RowenWoo committed:
[StarRocks On ES][Bug-Fix] handle limit operator push-down correctly (#9226)

if all conjunct can not processed by ES, limit operator can not be push down

RowenWoo committed:
[StarRocks On ES] generate match_all query when all predicates cannot push-down (#9208)

if all predicate can not push down , match_all query would generated

xiaoyong-z committed:
[BugFix] shouldn't close file system when broker encounters excecption (#9507)

In current implementation, broker will close file system when it encounters exception.
Suppose there are two threads, threads A is in the middle of reading process, thread B
is opening the file and it finds the file doesn't exist, then broker will throw an exception
and close the file system. However, thread A is still using the file system to read files,
close the file system will also lead to thread A abort reading.

rickif committed:
[Enhancement] Unable to filter NULL data by WHERE to build NOT NULL column (#9444)

The previous implementation checks whether the not-null column with null data before the predicate evaluation, which leads we cannot filter null record by setting WHERE KEY IS NOT NULL in load.
The new implementation removes the check before predicate evaluation.

murphyatwork committed:
[Feature] turn on the enable_column_expr_predicate by default (#9377)

stdpain committed:
[Refactor] Refactor "find next nozero index" in Column::filter_range (#9508)

shshenhua committed:
[SR SQL Planner Summer Camp] Support RoutineLoadStop Statement in new Parser and Analyzer (#8985)

zddr committed:
[BugFix] Forbid creating materialized view using week/quarter/year to refresh (#9465)

waittttting committed:
[Enhancement] Disable adding fe with the same host (#9503)

If we add a FE to the cluster with the same IP and a different edit log port, the FE can be added to the cluster under the current logic, but this situation will result in a series of errors.
This PR will fix this issue.
A new FE node will be blocked from joining the cluster as long as it has the same host as one of the existing nodes.

sduzh committed:
Fix NPE (#9539)

For LakeTable, getRelatedMaterializedViews() is null after FE restarted.
Since LakeTable does not support materialized view now, it's ok to remove
these codes.

silverbullet233 committed:
Fix error sql generated for jdbc oracle table (#9510)

xiaoyong-z committed:
[Feature] Support select into outfile without broker (#9201)

Linkerist committed:
[BugFix] Vectical rowset writer final flush need writer finalize footer for partial columns. (#9075)

Once the following conditions are met, a data load will cause the final merge in the rowset writer.
1. partial update
2. large data load in one batch to cause final merge
3. the number of partial columns large than config::vertical_compaction_max_columns_per_group [default = 5]

final merge will do vertical write under some circumstances, will clear partial_rowset_footers array at first, need to create new partial_rowset_footers for newly merged segments, old code of horizontal rowset writer did this, but the newly added VerticalRowsetWriter didn't, which will cause the partial_rowset_footers is empty. null pointer error will occur in rowset load like this:
```
#0  0x0000000001f8f7a0 in starrocks::BetaRowset::do_load() ()
#1  0x0000000001f8790c in starrocks::Rowset::load() ()
#2  0x0000000001f8e1d6 in starrocks::BetaRowset::get_segment_iterators2(starrocks::vectorized::Schema const&, starrocks::KVStore*, long, starrocks::OlapReaderStatistics*) ()
#3  0x0000000001c40c68 in starrocks::RowsetUpdateState::_do_load(starrocks::Tablet*, starrocks::Rowset*) ()
#4  0x0000000001c41775 in void std::call_once<starrocks::RowsetUpdateState::load(starrocks::Tablet*, starrocks::Rowset*)::{lambda()#1}>(std::once_flag&, starrocks::RowsetUpdateState::load(starrocks::Tablet*, starrocks::Rowset*)::{lambda()#1}&&)::{lambda()#2}::_FUN() ()
#5  0x00007fd681f9f1cb in __pthread_once_slow () from /lib64/libpthread.so.0
#6  0x0000000001c3d9b3 in starrocks::RowsetUpdateState::load(starrocks::Tablet*, starrocks::Rowset*) ()
#7  0x0000000001ad7203 in starrocks::UpdateManager::on_rowset_finished(starrocks::Tablet*, starrocks::Rowset*) ()
#8  0x00000000032ff75b in starrocks::vectorized::DeltaWriter::commit() ()
#9  0x00000000032f03c1 in starrocks::vectorized::AsyncDeltaWriter::_execute(void*, bthread::TaskIterator<starrocks::vectorized::AsyncDeltaWriter::Task>&) ()
#10 0x00000000040581ec in bthread::ExecutionQueueBase::_execute(bthread::TaskNode*, bool, int*) ()
#11 0x0000000004058fb8 in bthread::ExecutionQueueBase::_execute_tasks(void*) ()
#12 0x00000000021257e9 in starrocks::ThreadPool::dispatch_thread() ()
#13 0x000000000212139a in starrocks::Thread::supervise_thread(void*) ()
#14 0x00007fd681fa0e65 in start_thread () from /lib64/libpthread.so.0
#15 0x00007fd6815bb88d in __libc_ifunc_impl_list () from /lib64/libc.so.6
#16 0x0000000000000000 in ?? ()
```

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000000001f8f7a0 in google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<starrocks::FooterPointerPB>::TypeHandler> (index=0, this=<optimized out>) at /root/starrocks/be/../gensrc/build/gen_cpp/olap_file.pb.h:4467
4467	/root/starrocks/be/../gensrc/build/gen_cpp/olap_file.pb.h: No such file or directory.
[Current thread is 1 (LWP 51456)]
(gdb) bt
#0  0x0000000001f8f7a0 in google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<starrocks::FooterPointerPB>::TypeHandler> (index=0, this=<optimized out>) at /root/starrocks/be/../gensrc/build/gen_cpp/olap_file.pb.h:4467
#1  google::protobuf::RepeatedPtrField<starrocks::FooterPointerPB>::Get (index=0, this=<optimized out>) at /var/local/thirdparty/installed/include/google/protobuf/repeated_field.h:2173
#2  starrocks::RowsetTxnMetaPB::_internal_partial_rowset_footers (index=0, this=<optimized out>) at /root/starrocks/be/../gensrc/build/gen_cpp/olap_file.pb.h:4463
#3  starrocks::RowsetTxnMetaPB::partial_rowset_footers (index=0, this=<optimized out>) at /root/starrocks/be/../gensrc/build/gen_cpp/olap_file.pb.h:4467
#4  starrocks::RowsetMeta::partial_rowset_footer (segment_id=0, this=0x11ccc910) at /root/starrocks/be/src/storage/rowset/rowset_meta.h:174
#5  starrocks::BetaRowset::do_load (this=0xfd993f0) at /root/starrocks/be/src/storage/rowset/beta_rowset.cpp:81
#6  0x0000000001f8790c in starrocks::Rowset::load (this=this@entry=0xfd993f0) at /root/starrocks/be/src/storage/rowset/rowset.cpp:50
#7  0x0000000001f8e1d6 in starrocks::BetaRowset::get_segment_iterators2 (this=this@entry=0xfd993f0, schema=..., meta=meta@entry=0x0, version=version@entry=0, stats=stats@entry=0x7fd6619bc070) at /root/starrocks/be/src/storage/rowset/beta_rowset.cpp:358
#8  0x0000000001c40c68 in starrocks::RowsetUpdateState::_do_load (this=0x10576d50, tablet=0xf130380, rowset=0xfd993f0) at /root/starrocks/be/src/storage/rowset_update_state.cpp:81
#9  0x0000000001c41775 in operator() (__closure=0x7fd6619bc2b0) at /root/starrocks/be/src/storage/rowset_update_state.cpp:38
#10 std::__invoke_impl<void, starrocks::RowsetUpdateState::load(starrocks::Tablet*, starrocks::Rowset*)::<lambda()> > (__f=...) at /usr/include/c++/10.3.0/bits/invoke.h:60
#11 std::__invoke<starrocks::RowsetUpdateState::load(starrocks::Tablet*, starrocks::Rowset*)::<lambda()> > (__fn=...) at /usr/include/c++/10.3.0/bits/invoke.h:95
#12 operator() (this=<optimized out>) at /usr/include/c++/10.3.0/mutex:717
#13 operator() (this=0x0) at /usr/include/c++/10.3.0/mutex:722
#14 _FUN () at /usr/include/c++/10.3.0/mutex:722
#15 0x00007fd681f9f1cb in __pthread_once_slow () from /lib64/libpthread.so.0
#16 0x0000000001c3d9b3 in __gthread_once (__func=<optimized out>, __once=0x10576d50) at /usr/include/c++/10.3.0/x86_64-pc-linux-gnu/bits/gthr-default.h:700
#17 std::call_once<starrocks::RowsetUpdateState::load(starrocks::Tablet*, starrocks::Rowset*)::<lambda()> > (__f=..., __once=...) at /usr/include/c++/10.3.0/mutex:729
#18 starrocks::RowsetUpdateState::load (this=this@entry=0x10576d50, tablet=tablet@entry=0xf130380, rowset=rowset@entry=0xfd993f0) at /root/starrocks/be/src/storage/rowset_update_state.cpp:36
#19 0x0000000001ad7203 in starrocks::UpdateManager::on_rowset_finished (this=0x94a3a40, tablet=0xf130380, rowset=0xfd993f0) at /root/starrocks/be/src/storage/update_manager.cpp:260
#20 0x00000000032ff75b in starrocks::vectorized::DeltaWriter::commit (this=this@entry=0xfc5f790) at /root/starrocks/be/src/storage/storage_engine.h:164
#21 0x00000000032f03c1 in starrocks::vectorized::AsyncDeltaWriter::_execute (meta=0xfc5f790, iter=...) at /root/starrocks/be/src/storage/async_delta_writer.cpp:38
#22 0x00000000040581ec in bthread::ExecutionQueueBase::_execute (this=0xabbe600, head=<optimized out>, high_priority=<optimized out>, niterated=0x0) at /var/local/thirdparty/src/incubator-brpc-0.9.7/src/bthread/execution_queue.cpp:272
#23 0x0000000004058fb8 in bthread::ExecutionQueueBase::_execute_tasks (arg=<optimized out>) at /var/local/thirdparty/src/incubator-brpc-0.9.7/src/bthread/execution_queue.cpp:151
#24 0x00000000021257e9 in std::function<void ()>::operator()() const (this=<optimized out>) at /usr/include/c++/10.3.0/bits/std_function.h:248
#25 starrocks::FunctionRunnable::run (this=<optimized out>) at /root/starrocks/be/src/util/threadpool.cpp:44
#26 starrocks::ThreadPool::dispatch_thread (this=0xac36540) at /root/starrocks/be/src/util/threadpool.cpp:513
#27 0x000000000212139a in std::function<void ()>::operator()() const (this=0x9e02c58) at /usr/include/c++/10.3.0/bits/std_function.h:248
#28 starrocks::Thread::supervise_thread (arg=0x9e02c40) at /root/starrocks/be/src/util/thread.cpp:326
#29 0x00007fd681fa0e65 in start_thread () from /lib64/libpthread.so.0
#30 0x00007fd6815bb88d in __libc_ifunc_impl_list () from /lib64/libc.so.6
#31 0x0000000000000000 in ?? ()
```

Astralidea committed:
[Feature] Support modify mysql server version (#9518)

At present, several community users have reported that the version of mysql server we display is too low, and it will be considered to be vulnerable by security scans.
Expect to upgrade to a higher version like provide configuration that can be modified.

liuyehcf committed:
[Feature] Support correlated scalar non-agg subquery (#9152)

Youngwb committed:
[UT] Enable Cte Reuse for UT (#9524)

predator4ann committed:
[Feature] New Parser and Analyzer support ShowLoadStmt/ShowLoadWarningsStmt/CancelLoadStmt (#8825)

ucasfl committed:
[Feature] Create history partitions automatically when start is specified for dynamic partition (#9501)

```sql
CREATE TABLE test
(
    `day_` DATE,
    `hour_` DATETIME,
    `minute_` DATETIME,
    `second_` DATETIME, 
    `timestamp_` INT,
    `nickname` VARCHAR(65533)
)ENGINE=OLAP
PRIMARY KEY(`day_`)
PARTITION BY RANGE(`day_`)()
DISTRIBUTED BY HASH(`day_`) BUCKETS 128 
PROPERTIES (
"replication_num" = "2",
"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "Day",
"dynamic_partition.time_zone" = "Asia/Shanghai",
"dynamic_partition.start" = "-7",
"dynamic_partition.end" = "3",
"dynamic_partition.prefix" = "p",
"dynamic_partition.buckets" = "128",
"dynamic_partition.create_history_partition" = "true",
"dynamic_partition.history_partition_num" = "7",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
```
When create table, we can use properties of `dynamic_partition.create_history_partition` and `dynamic_partition.history_partition_num` to create history partitions automatically.

caneGuy committed:
[Bugfix] Fix max src rssid error during update compaction (#9492)

Commit:
[Doc] Add 2.2.4 Release notes (#9549)

mxdzs0612 committed:
[Refactor] Rename ICEBERG_CATALOG to iceberg.catalog.type (#9451)

Rename the property ICEBERG_CATALOG from starrocks.catalog-type to iceberg.catalog.type
This refactor will take effects on current iceberg users. The document and daily test need to be modified, too.

sduzh committed:
[BugFix] Fix potential bug and refine log printing (#9236)

sevev committed:
Change keys parameter type to Slice (#9547)

Change keys parameter type of PersistentIndex to Slice. This pr for the subsequent simplification of the interface to support varlen key implementation

GavinMar committed:
[BugFix] Fix the failure of querying extenal table that contains struct columns (#9420)

When external hive table contains map or struct columns, querying may
failed for mismatched shared buffer offset. Because the shared buffer
stream does not locate the column meta data correctly when
setting io range and we should use physical column index as the meta
index.

wyb committed:
[Lake] support read and compact lake tablet with delete rowset (#9398)

1. Add delete predicate processing in lake tablet reader and compaction policy.
2. Fix update cumulative point bug in base compaction.

sduzh committed:
[Refactor] Rename DropTabletRequest to DeleteTabletRequest (#9562)

wyb committed:
[Lake] Support lake tablet statistic collection (#9460)

1. FE periodically collect lake tablet statistic, such as num rows and data size.
2. Num rows statistic is used in CBO optimizer.
3. Data size statistic is used in show command, like show data.

decster committed:
[Enhancement] PseudoBackend support report task; add failure injection and write quorum test (#9489)

The PR adds task report functionality to PseudoBackend, and adds write failure injection to test txn's quorum write logic.

Astralidea committed:
[Bugfix] Fix NPE when refresh mv does not exist (#9560)

At this time, because the materialized view does not exist when it is refreshed.

meegoo committed:
[BugFix] Fix parquet array convert bug (#9402)

meegoo committed:
[BugFix] Fix FileScanNode use not null column (#9546)

FileScanNode use dest table's column type, it may be not nullable. But after data scan, it may be convert to nullable column.
It will make schema different between each chunk, which will make exchange node crash since it cache the chunk schema.

Youngwb committed:
[BugFix] Fix Prune Distinct Global Aggregate lose projection (#9548)

blackstar-baba committed:
[Bugfix] There is no data when use cte (#9567)

sduzh committed:
[Refactor] Remove unnecessary move (#9582)

Moving a temporary object prevents copy elision

sevev committed:
[Enhancement] Support varlen key for ImmutableIndex (#9495)

Change keys parameter type of PersistentIndex to Slice. This pr for the subsequent simplification of the interface to support varlen key implementation

sduzh committed:
[Refactor] Remove unnecessary move (#9584)

Moving a temporary object prevents copy elision

Astralidea committed:
[Bugfix] add system variable 'transaction_isolation' (#9586)

Youngwb committed:
[BugFix] record db name of view when query dump (#9570)

adzfolc committed:
[Feature]Hudi connector (#8614)

rickif committed:
[BugFix] Show routine load for non-existed job (#9572)

chaoyli committed:
[Refactor] Remove the AlterTask in TabletMeta. (#9561)

liuyehcf committed:
[BugFix] Fix unknown error of profile merge mechanism (#9556)

padmejin committed:
[BugFix] Discard expired backup/restore job (#9531)

padmejin committed:
[BugFix] Remove colocated index from memory (#9578)

* [BugFix] remove colocate index from memory

* Update fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java

* fix ut

kangkaisen committed:
Add Checklist to PULL_REQUEST_TEMPLATE (#9599)

Astralidea committed:
Fix delete throw unknown error when enable query detail (#9597)

This because if FE config enable_collect_query_detail_info = true
in StmtExecutor context.getQueryDetail != null will apply and will call buildExplaianString
but marked delete does not have explainString so execPlan will be null cause of NPE.

nshangyiming committed:
[Enhancement] Schedule VERSION_INCOMPLETE task imediately (#9590)

ZiheLiu committed:
[BugFix] Synchronize cpu_cores of BEs when replaying master FE info to other FEs (#9580)

dulong41 committed:
[Actions] source env.sh to run run-fe-ut.sh (#9594)

* Update sonar4fe.yml

* Update sonar4fe.yml

QingdongZeng3 committed:
[Bugfixi] not support create mv by mv #9543 (#9576)

[Bugfix] not support create mv by mv #9543

dulong41 committed:
[Actions] mark stale for unnoticed issues and pull requests (#9526)

* mark stale for unnoticed issues and pull requests

* Update stale.yml

* Update stale.yml

Co-authored-by: dulong <dulong@starrocks.com>

sduzh committed:
Clear storage on drop table (#9559)

melt-code committed:
[Refactor] support backup statement in new analyzer (#8926)

Seaven committed:
[BugFix] Fix multi nest cte inline error (#9467)

murphyatwork committed:
optimize performance of json_query functions (#9614)

Optimize the performance of `json_query/get_json_string/get_json_int` functions:
- Improvement: 10B rows of data, 16 `get_json_string` expression in select clause; improve from `12s` to `6s`
- Inline the type convert into `_json_query_impl` function, reduce the overhead of construct immediate Json object
- Specialize `get_json_string`, avoid call `Slice::toJson` but read the string directly

HangyuanLiu committed:
[Enhancement] Modify the sampling strategy for histogram statistics #9505

predator4ann committed:
[Feature] New Parser and Analyzer support ModifyBrokerClause (#8815)

smartlxh committed:
[Bugfix] mv task info not exists after modify async to MANUAL (#9620)

The PERIODICAL task has been droped ,but there is not a EVENT_TRIGGERED task

liuyehcf committed:
[Bugfix] Fix ConcurrentModificationException of concurrently accessing childCounterMap (#9621)

packy92 committed:
[BugFix] build script supports Python3 and fix file path typos (#9563) (#9571)

Add a default python execute command property and update it when user set a PYTHON environment variable. Use this property in antrun-plugin to ensure build successfully in platform with Python2.x installed or PYTHON environment variable had been set. This won't help user who only installed Python3 and without setting PYTHON environment variable.

padmejin committed:
[BugFix] Close invalid environment before bdb setup retry (#9427)

* [BugFix] adjust retry policy: close environment if is valid

decster committed:
[Bugfix] disable txngraph if new publish is not enabled (#9617)

ucasfl committed:
add java format style (#9615)

Youngwb committed:
[BugFix] Fix correlated scalar non-aggregate subquery unknown error when correlated predicate has expression (#9629)

padmejin committed:
[BugFix] fix `SHOW PROC /colocation_group' after erase db with same name (#9604)

In my previous implementation, when executing SHOW PROC /colocation_group, there will be a star next to the table id if the ID of the database / table is in CatalogRecyleBin.
If we create & drop databases with the same name twice, the first database will lose its Database ID from CatalogRecycleBin. As a result, it will look like a normal table in the output.
To fix this, now we only validate if the ID of the database or table in GlobalStateMgr.

sduzh committed:
[Enhancement][Lake] Delete segment files on abort txn (#9603)

xiaoyong-z committed:
[Feature] Support spark load without broker (#9228)

Signed-off-by: xyz <a997647204@gmail.com>

LiShuMing committed:
[Enhancement] Optimize hash join when hash table is empty (#9606)

nshangyiming committed:
[BugFix] Fix not enough hosts error when BE is decommissioned (#9473)

1. When multiple BEs reside on the same host, only one BE will be chosen when create table. But if some of them are decommissioned, FE still might choose the decommissioned one, this will cause the problem described in the issue.
2. Also refine some codes based on sonar code smell.

dirtysalt committed:
[Enhancement] add more error info to s3 sdk (#9650)

nshangyiming committed:
[Enhancement] Avoid VERSION_INCOMPLETE repair task got blocked (#9573)

We should create a REPLICA_MISSING type task only when there exists enough available BEs which
we can choose to clone data. If not, we should check if we can create a VERSION_INCOMPLETE task
so that the repair of the replica with an incomplete version won't be blocked. Therefore the version publish process
of load task won't be blocked either.

decster committed:
[Enhancement] Add clone logic to PseudoCluster and UT for tablet repair and clone (#9607)

stdpain committed:
[Bug] fix crc32 heap-buffer-overflow in aarch64 (#9669)

amber-create committed:
Add v2.0.9 release notes (#9689)

LiShuMing committed:
[Bugfix] fix coredump when hash table is empty (#9678)

GavinMar committed:
[Bugfix] Fix the redundant read request caused by mismatched buffer offset for parquet. (#9682)

When reading parquet page header, the default buffer stream increases page offset less than buffer postion, which causes the data buffer to be reset when reading data page, and and the data previously read is discarded.

This triggers more remote read requests, resulting in reading performance degradation.

sduzh committed:
[Refactor]Minor: refactor SchemaChangeHandler (#9635)

LiShuMing committed:
Revert "[Enhancement] Optimize hash join when hash table is empty" (#9701)

trueeyu committed:
Remove auto register config from MemTracker (#9668)

sduzh committed:
[Refactor] Remove LakeServiceClient (#9686)

meegoo committed:
[Enhancement] Support load_transmission_compression_type session variable unified load compress config (#9422)

nshangyiming committed:
[BugFix] Fix recovery for bad replica (#9666)

I will add test case using pseduo cluster (#8854) in another pr, because there are some functionalities about tablet schedule needed to add in pseudo cluster first.

xiaoyong-z committed:
[BugFix] Null date and datetime value for spark load is not consistent (#9598)

* [BugFix] null date and datetime value for spark load is not consistent with broker/stream load

For broker/stream load, null value for date and datetime is DefaultValueGenerator<DateValue/TimestampValue>,
however, for spark load, it is not so.
So when we create a aggregate table, we first perform broker load, and then perform spark load on the same data
(here assuming the data has null value), and query by select count(*) from table, the result will contain two null value.

In this pr, we let the null date and datetime value for spark load to be consistent with broker/stream load

HangyuanLiu committed:
Change the statistics table to non-static to prevent CreateAnalyzer from modifying ColumnDef (#9396)

HangyuanLiu committed:
Statistics collection focus on truncate and insert overwrite (#9699)

padmejin committed:
[Feature] Support all-digit user name (#9715)

Allow all-digit user name, such as create user '12345';.

Astralidea committed:
Add delete query detail Case for StmtExecutor (#9727)

The PR related to this case is [Bugfix] Fix delete throw unknown error when enable query detail #9597

dulong41 committed:
[Actions] stale actions notification update (#9730)

* Update stale.yml

* Update stale.yml

gengjun-git committed:
[BugFix] Fix BrokerLoad FailMsg serialization throws NPE (#9728)

This bug appears in a customer's environment that has an internal version based on 2.1.2. But there are too many callers of FailMsg(CancelType cancelType, String msg); we should write some protectable code to prevent the NPE.

HangyuanLiu committed:
[BugFix] Fix set statement miss user define variable and system variable syntax bug（2/2) (#9718)

Linkerist committed:
[Refactor] Remove rowset writer factory and storage_format_v1 (#9565)

smartlxh committed:
[Feature] Support load json type from csv format through stream load (#9520)

Support load json type from csv format through stream load by adding a json_converter

liuyehcf committed:
[Bugfix] Fix incorrect partition property (#9622)

silverbullet233 committed:
[Enhancement] introduce a lock-free queue (#9711)

stdpain committed:
[Bugfix] fix miss string function in MAX/MIN function when enable dictionary optimization (#9632)

we will add a new dictionary column for MAX/MIN aggregate function in dictionary optimization.
so when we have a string function for a old dictionary column, we also need apply it to the new dictionary column.

abc982627271 committed:
fix ut
